### ISSUE TITLE
Syntax error when executing transaction with go pq sql driver library.

### ISSUE THREAD

**Issue by andrewchambers** [cockroachdb/cockroach#16309]
**BUG REPORT**

1. Please supply the header (i.e. the first few lines) of your most recent
   log file **for each node in your cluster**.

I170603 06:28:35.309149 61 util/log/clog.go:908  [config] file created at: 2017/06/03 06:28:35
I170603 06:28:35.309149 61 util/log/clog.go:908  [config] running on machine: cockroach-1
I170603 06:28:35.309149 61 util/log/clog.go:908  [config] binary: CockroachDB CCL v1.0 (linux amd64, built 2017/05/10 13:19:57, go1.8.1)
I170603 06:28:35.309149 61 util/log/clog.go:908  [config] arguments: [cockroach start --insecure --store /cockroachdata]

2. Please describe the issue you observed:

- What did you do?

Ran a transaction with github.com/lib/pq and github.com/lib/pq

- What did you expect to see?

The transaction to execute correctly.

- What did you see instead?

```
2017/06/04 10:28:54 an error occured during billing: error billing due account 247821695300435969: pq: syntax error at or near "read"
BEGIN READ WRITE
```

The issue seems to be cockroachdb does not support the READ WRITE syntax used by the pq driver at the following line:

https://github.com/lib/pq/blob/8837942c3e09574accbc5f150e2c5e057189cace/conn_go18.go#L68

**Comment by benesch** [comment]
Hi @andrewchambers—thanks for reporting! Looks like [this change was made 4 days ago](https://github.com/lib/pq/commit/ed1a26d9e1fd4eb1cc491e6d688467f4e81a413c) to lib/pq and our Postgres-compatibility layer hasn't caught up yet.

Assigning to @mjibson, since he approved these changes to lib/pq 😃.

**Comment by mjibson** [comment]
@tamird this is caused because the new sql options add either "READ ONLY" or "READ WRITE" to the "BEGIN" statement instead of defaulting to the empty string if read only is false. I think we should add a READ ONLY/WRITE to the cockroach grammar (and have read only error, and read write be ignored) instead of change lib/pq. We already do similar stuff for the isolation and priority levels. Thoughts?

**Comment by benesch** [comment]
(That sounds right to me, FWIW!)

**Comment by tamird** [comment]
Yep, agreed.

**Comment by petermattis** [comment]
We should probably support `READ ONLY` transactions. Should be straightforward.

**Comment by jordanlewis** [comment]
#16033 is another related feature request, for setting a transaction to be read only.

**Comment by mjibson** [comment]
Fixed by #16387
